pkg/k8shandler/collection.go

```diff
@@ -105,6 +157,66 @@ func createOrUpdateFluentdConfigMap(logging *logging.ClusterLogging) error {
 	return nil
 }
 
+func createOrUpdateRsyslogConfigMap(logging *logging.ClusterLogging) error {
+
+	// need three configmaps
+	// - one for rsyslog run.sh script - rsyslog-bin
+	// - one for main rsyslog.conf file - rsyslog-main
+	// - one for the actual config files - rsyslog
+	rsyslogConfigMaps := make(map[string]*v1.ConfigMap)
+	rsyslogBinConfigMap := utils.ConfigMap(
+		"rsyslog-bin",
+		logging.Namespace,
+		map[string]string{
+			"rsyslog.sh": string(utils.GetFileContents("files/rsyslog/rsyslog.sh")),
```

Comment:
I was actually wondering why we had the `rsyslog.sh` file there. Is there a reason this isn't baked into the image instead?

Reply:
Yes - the image, including its built-in run script, is designed to be used as a replacement for the system rsyslog.  Our requirements are very different, so we should provide our own run.sh script.  Basically, we are treating the container as a bag of packages and a runtime, with no scripts or configuration for us to use - we will provide/replace all scripts/configuration in the image.